go/cmd/dolt/commands/push.go

```diff
@@ -392,6 +392,11 @@ func pushToRemoteBranch(ctx context.Context, dEnv *env.DoltEnv, mode ref.RefUpda
 				cli.Println("hint: 'dolt pull ...') before pushing again.")
 				return errhand.BuildDError("").Build()
 			} else {
+				errString := err.Error()
+				if errString == "rpc error: code = PermissionDenied desc = permission denied" {
```

Comment:
Is there a better way to match this error?

Reply:
You should be able to match RPC errors with this method: https://godoc.org/google.golang.org/grpc/status#FromError

Reply:
Fixed https://github.com/dolthub/dolt/pull/1047/commits/6089d7e15d5fe4b02a4dc13630289baee7f937b0